pipelines/batch/src/main/java/com/google/fhir/analytics/JdbcResourceWriter.java

```diff
@@ -196,18 +197,22 @@ private static void deleteRowsById(DataSource dataSource, String tableName, Stri
    *
    * @param resourceType the type of resource to be deleted
    * @param id the id of the resource to be deleted
-   * @throws SQLException
+   * @throws SQLException if a database access error occurs
    */
   public void deleteResourceById(String resourceType, String id) throws SQLException {
     if (viewManager == null) {
       deleteRowsById(jdbcDataSource, resourceType, id);
     } else {
       ImmutableList<ViewDefinition> views = viewManager.getViewsForType(resourceType);
-      for (ViewDefinition vDef : views) {
-        if (Strings.isNullOrEmpty(vDef.getName())) {
-          throw new SQLException("Field `name` in ViewDefinition is not defined.");
+      if (views != null) {
```

Comment:
I think this is one of those examples that it would have made more sense to return an empty list (instead of `null`) in `getViewsForType`; and then get rid of many null checks. It would be nice if you make that change.